ruby-3.3.yaml

```diff
@@ -88,42 +99,74 @@ pipeline:
+      rm -r ${RUBYDIR}/bundler
+      rm ${RUBYDIR}/bundler.rb
 
-      GEMDIR=${{targets.destdir}}${{vars.prefix}}/lib/ruby/gems/$RUBYWILD
-      rm -rf ${GEMDIR}/gems/bundler-*
-      rm -f ${GEMDIR}/specifications/default/bundler-*.gemspec
+      GEMDIR=${{targets.destdir}}/usr/lib/ruby/gems/${{vars.stdlib-version}}
+      rm -r ${GEMDIR}/gems/bundler-*
+      rm ${GEMDIR}/specifications/default/bundler-*.gemspec
 
-      rm -f ${{targets.destdir}}/usr/bin/bundle \
-         ${{targets.destdir}}/usr/bin/bundler
+      rm ${{targets.destdir}}/usr/libexec/ruby/${{vars.stdlib-version}}/bundle \
+         ${{targets.destdir}}/usr/libexec/ruby/${{vars.stdlib-version}}/bundler
 
   - runs: |
       find ${{targets.destdir}} -name 'mkmf.log' -exec rm {} \;
 
 subpackages:
-  - name: "ruby-3.3-doc"
-    description: "ruby documentation"
+  - name: "${{package.name}}-base"
+    description: "Ruby ${{vars.major-minor-version}} base"
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.contextdir}}
+
+          mv ${{targets.destdir}}/* ${{targets.contextdir}}
+
+          mkdir -p ${{targets.destdir}}/usr/bin
+          mkdir -p ${{targets.contextdir}}/usr/bin
+
+          for bin in ${{targets.contextdir}}/usr/libexec/ruby/${{vars.stdlib-version}}/*; do
+            bn="$(basename $bin)"
+            ln -s /usr/libexec/ruby/${{vars.stdlib-version}}/$bn ${{targets.destdir}}/usr/bin/
+            ln -s /usr/libexec/ruby/${{vars.stdlib-version}}/$bn ${{targets.contextdir}}/usr/bin/$bn\${{vars.major-minor-version}}
+          done
+
+  - name: "${{package.name}}-doc"
+    description: "Ruby ${{vars.major-minor-version}} documentation"
     pipeline:
-      - uses: split/manpages
       - runs: |
-          mkdir -p "${{targets.subpkgdir}}"/usr/share
-          mv "${{targets.destdir}}"/usr/share/doc "${{targets.subpkgdir}}"/usr/share/
-          mv "${{targets.destdir}}"/usr/share/ri "${{targets.subpkgdir}}"/usr/share/
+          datadir="/usr/share/ruby/${{vars.stdlib-version}}"
 
-  - name: "ruby-3.3-dev"
-    description: "ruby development headers"
+          mkdir -p "${{targets.contextdir}}/$datadir"
+
+          mv "${{targets.outdir}}/${{package.name}}-base/$datadir/doc" "${{targets.contextdir}}/$datadir/"
+          mv "${{targets.outdir}}/${{package.name}}-base/$datadir/man" "${{targets.contextdir}}/$datadir/"
+          mv "${{targets.outdir}}/${{package.name}}-base/usr/share/ri" "${{targets.contextdir}}/usr/share/"
+
+  - name: "${{package.name}}-base-dev"
+    description: "Ruby ${{vars.major-minor-version}} base development headers"
+    dependencies:
+      runtime:
+        - ${{package.name}}-base=${{package.full-version}}
     pipeline:
       - uses: split/dev
+        with:
+          package: "${{package.name}}-base"
     test:
       pipeline:
         - uses: test/pkgconf
 
+  # Empty package, ensures ruby-dev always installs development files for latest ruby version
+  - name: "${{package.name}}-dev"
```

Comment:
This is fine, but I think we are generally trying to move away from "virtual provides".
https://github.com/wolfi-dev/os/issues/32874 describes some reasons.  Those reasons do not completely line up with ruby-dev (using full-version here is _fine_, but using full-version in a ruby-pkgname is _not_ fine), but the point "virtual provides are only for humans" is relevant here.

So, i'm fine to keep this here, but I think we probably want to move away from virtual-provides like this.

Reply:
I wasn't actually aware that is a problem with virtual provides. I had always assumed that the highest version always took precedence. Completely agree that these should never be used when building images or as explicit dependencies in packages